### ISSUE TITLE
'RORPath instance has no attribute 'path'' raised of class '<type 'exceptions.AttributeError'>'

### ISSUE THREAD

**Issue by isalgueiro** [rdiff-backup/rdiff-backup#14]
Got this exception executing `rdiff-backup --no-acls`

```
Exception 'RORPath instance has no attribute 'path'' raised of class '<type 'exceptions.AttributeError'>':
  File "rdiff_backup\Main.pyc", line 304, in error_check_Main
  File "rdiff_backup\Main.pyc", line 324, in Main
  File "rdiff_backup\Main.pyc", line 280, in take_action
  File "rdiff_backup\Main.pyc", line 343, in Backup
  File "rdiff_backup\backup.pyc", line 51, in Mirror_and_increment
  File "rdiff_backup\backup.pyc", line 243, in patch_and_increment
  File "rdiff_backup\rorpiter.pyc", line 284, in __call__
  File "rdiff_backup\backup.pyc", line 718, in start_process

Traceback (most recent call last):
  File "rdiff-backup", line 30, in <module>
  File "rdiff_backup\Main.pyc", line 304, in error_check_Main
  File "rdiff_backup\Main.pyc", line 324, in Main
  File "rdiff_backup\Main.pyc", line 280, in take_action
  File "rdiff_backup\Main.pyc", line 343, in Backup
  File "rdiff_backup\backup.pyc", line 51, in Mirror_and_increment
  File "rdiff_backup\backup.pyc", line 243, in patch_and_increment
  File "rdiff_backup\rorpiter.pyc", line 284, in __call__
  File "rdiff_backup\backup.pyc", line 718, in start_process
AttributeError: RORPath instance has no attribute 'path'
```

I'm using rdiff-backup-1.2.8 on Windows 10.

**Comment by isalgueiro** [comment]
I no longer use rdiff-backup.

**Comment by mhsmith** [comment]
I've experienced this error as well in the past, though I didn't make a note of exactly what I was doing at the time. Please reopen the issue, at least until it's received a response from the developer.

**Comment by anomaly256** [comment]
Re-opening so we know to investigate it.  @mhsmith how did you get around it at the time?

**Comment by mhsmith** [comment]
I'm sorry, I don't remember. All I can tell you is I was running on Cygwin on Windows 7, and probably wasn't using any command-line options except `--exclude-globbing-filelist`. I don't think I've ever used `--no-acls`.

**Comment by maffe** [comment]
I got this error message when I ran `sudo rdiff-backup …` instead of doing `sudo su` and then `rdiff-backup …`.

**Comment by mhsmith** [comment]
Today I tried to use the build from http://savannah.nongnu.org/download/rdiff-backup/rdiff-backup-1.2.8-win32.zip rather than the build provided by Cygwin, and I received this exact error. 

Before that, I received the following messages:
```
Warning: Windows Access Control List file not found
UpdateError: <filename> changed from regular file before signature 
[this line was repeated several times, always for files with non-ASCII filenames]
```

I did not find any workaround other than going back to the Cygwin build.

**Comment by ymartin59** [comment]
I faced same error message:
```
sudo ./bin/backup.sh
[sudo] password for user: 
Run rdiff-backup on /home/user...
Exception 'RORPath instance has no attribute 'path'' raised of class '<type 'exceptions.AttributeError'>':
  File "/usr/lib/python2.7/dist-packages/rdiff_backup/Main.py", line 304, in error_check_Main
    try: Main(arglist)
  File "/usr/lib/python2.7/dist-packages/rdiff_backup/Main.py", line 324, in Main
    take_action(rps)
  File "/usr/lib/python2.7/dist-packages/rdiff_backup/Main.py", line 280, in take_action
    elif action == "backup": Backup(rps[0], rps[1])
  File "/usr/lib/python2.7/dist-packages/rdiff_backup/Main.py", line 343, in Backup
    backup.Mirror_and_increment(rpin, rpout, incdir)
  File "/usr/lib/python2.7/dist-packages/rdiff_backup/backup.py", line 51, in Mirror_and_increment
    DestS.patch_and_increment(dest_rpath, source_diffiter, inc_rpath)
  File "/usr/lib/python2.7/dist-packages/rdiff_backup/backup.py", line 243, in patch_and_increment
    ITR(diff.index, diff)
  File "/usr/lib/python2.7/dist-packages/rdiff_backup/rorpiter.py", line 284, in __call__
    branch.start_process(*args)
  File "/usr/lib/python2.7/dist-packages/rdiff_backup/backup.py", line 718, in start_process
    ("Either %s or %s must be a directory" % (repr(diff_rorp.path),

Traceback (most recent call last):
  File "/usr/bin/rdiff-backup", line 30, in <module>
    rdiff_backup.Main.error_check_Main(sys.argv[1:])
  File "/usr/lib/python2.7/dist-packages/rdiff_backup/Main.py", line 304, in error_check_Main
    try: Main(arglist)
  File "/usr/lib/python2.7/dist-packages/rdiff_backup/Main.py", line 324, in Main
    take_action(rps)
  File "/usr/lib/python2.7/dist-packages/rdiff_backup/Main.py", line 280, in take_action
    elif action == "backup": Backup(rps[0], rps[1])
  File "/usr/lib/python2.7/dist-packages/rdiff_backup/Main.py", line 343, in Backup
    backup.Mirror_and_increment(rpin, rpout, incdir)
  File "/usr/lib/python2.7/dist-packages/rdiff_backup/backup.py", line 51, in Mirror_and_increment
    DestS.patch_and_increment(dest_rpath, source_diffiter, inc_rpath)
  File "/usr/lib/python2.7/dist-packages/rdiff_backup/backup.py", line 243, in patch_and_increment
    ITR(diff.index, diff)
  File "/usr/lib/python2.7/dist-packages/rdiff_backup/rorpiter.py", line 284, in __call__
    branch.start_process(*args)
  File "/usr/lib/python2.7/dist-packages/rdiff_backup/backup.py", line 718, in start_process
    ("Either %s or %s must be a directory" % (repr(diff_rorp.path),
AttributeError: RORPath instance has no attribute 'path'
```

And command is `/usr/bin/rdiff-backup --print-statistics --exclude-filelist exclude.log /home/user /media/backup/home-user`

And the reason it may have failed is that script was previously started as regular `user` and it was now the first time I run script with `sudo`.

**Comment by ericzolf** [comment]
The issue is still there but it does only happen when another issue happens (I guess wrong or unreachable directory). RORPath class has indeed no attribute `path` hence the lines around 783 in `src/rdiff/backup.py` are wrong:

```python
        assert diff_rorp.isdir() or self.base_rp.isdir(), \
          ("Either %s or %s must be a directory" % (repr(diff_rorp.path),
```

Replacing it with `diff_rorp.get_safeindexpath()` should solve the issue. I'll fix it once #60 is closed.
